src/Pages/AllTopics.jsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from "react";
+import TopicCard from "../Components/TopicCard";
+import { getTopics } from "../utils/api";
+
+export default function AllTopics() {
+  const [topics, setTopics] = useState({});
+  const [isloading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getTopics().then(({ topics }) => {
```

Comment:
When you use the loading state it's always good to set isLoading to true at the top of the useEffect as when the useEffect is triggered we're assuming data is being loaded again so we'd want to trigger the loading state to appear